redis/src/main/scala/zio/redis/transactional/RedisTransaction.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.transactional
+
+import zio.redis.Output.QueuedOutput
+import zio.redis.{Output, RedisCommand, RedisError}
+import zio.{ZIO, Zippable}
+
+sealed trait RedisTransaction[+Out] { self =>
+  import RedisTransaction._
+
+  def commit: ZIO[Redis, RedisError, Out] =
+    ZIO.serviceWithZIO[Redis] { redis =>
+      self match {
+        case Single(command, in) =>
+          command.run(in)
```

Comment:
Is it possible reuse the existing API here (e.g. pass by name), rather than having a dedicated package full of duplicates?